recipes/nnpdf/recipe.yaml

```
@@ -0,0 +1,78 @@
+schema_version: 1
+
+context:
+  name: nnpdf
+  version: 4.0.10
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  url: https://github.com/NNPDF/nnpdf/archive/${{ version }}.tar.gz
+  sha256: 307bc48d51b44ab8cf24e365d1a89bbd50f62290f9a3142eae5c55188b9cb49d
+
+build:
+  number: 0
+  noarch: python
+  script:
+    - export POETRY_DYNAMIC_VERSIONING_BYPASS=${{ version }}
+    - ${{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+
+requirements:
+  host:
+    - python =${{ python_min }}
+    - poetry-core >=1.0.0
+    - poetry-dynamic-versioning >=1.2.0
+    - pip
+  run:
+    - python >=${{ python_min }}
+    - tensorflow >=2.17
+    - keras >=3.1
+    - psutil
+    - hyperopt
+    - mongodb
+    - pymongo <4
+    - pyopenssl >=23.2
+    - seaborn
+    - lhapdf
+    - numpy
+    - pkg-config
+    - reportengine
+    - matplotlib >=3.9
+    - blessings >=1.7
+    - scipy >=0.19.1
+    - pandas
+    - requests
+    - prompt_toolkit
+    - validobj
+    - pineappl >=0.8.2,<1.0
+    - eko >=0.14.2,<0.15
+    - fiatlux
+    - joblib
+    - ruamel.yaml >=0.15
+
+tests:
+  - requirements:
+      run:
+        - hypothesis
+        - pytest
+        - pytest-mpl
+
+    script:
+      # This test requires to download stuff not available from here
+      - pytest --mpl --mpl-default-tolerance 24 --pyargs validphys -k 'not extrasum_slice'
+      - if: linux
+        then: # this test is too slow in macos
+          - pytest --pyargs n3fit
```

We only care if Linux passes here for noarch. We do want to know if the package is OK on macOS and Windows, we already know the former, let's run, even if it is slow, the tests on macOS at least once here in staged-recipes. In the feedstock we will build only Linux.

```suggestion
      # This test requires to download stuff not available from here
      - pytest --mpl --mpl-default-tolerance 24 --pyargs validphys -k 'not extrasum_slice'
      - pytest --pyargs n3fit
```